src/GameLogic.cpp

```diff
@@ -424,3 +446,16 @@ void GameLogic::enemyShoot(Enemy& enemy) {
 		actorsVector.push_back(enemy.getProjectile());
 	}
 }
+
+
+void GameLogic::addPreference(std::string pref, int type)
```

Comment:
I think it would be easier to have 4 booleans, one for each enemy type and then set them to true or false depending on whether that enemy type is good or bad. Making a vector and then adding to it seems like it would take a lot more overhead and require traversing through the vector every time you need to check whether an enemy is supposed to be killed or ignored. We can discuss whether you think that would actually make a difference or not